src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs

```diff
@@ -329,6 +364,26 @@ public void ApplyForwarders(HttpContext context)
                 }
                 request.Host = HostString.FromUriComponent(currentValues.Host);
             }
+
+            if (checkPrefix && currentValues.Prefix != null)
+            {
+                // Save the original
+                requestHeaders[_options.OriginalPrefixHeaderName] =
+                    request.PathBase.HasValue ? request.PathBase.ToString() : "/";
+
+                if (forwardedPrefix!.Length > entriesConsumed)
+                {
+                    // Truncate the consumed header values
+                    requestHeaders[_options.ForwardedPrefixHeaderName] = forwardedPrefix.Take(forwardedPrefix.Length - entriesConsumed).ToArray();
+                }
+                else
+                {
+                    // All values were consumed
+                    requestHeaders.Remove(_options.ForwardedPrefixHeaderName);
+                }
+
+                request.PathBase = PathString.FromUriComponent(currentValues.Prefix.TrimEnd('/'));
```

Comment:
TrimEnd shouldn't be needed, pass the given value through. PathString has built in logic to avoid duplicate slashes when combined with another PathString (PathBase + Path).
https://github.com/dotnet/aspnetcore/blob/a02b94e36c6b26da67a74115aeed7fd621b1365e/src/Http/Http.Abstractions/src/PathString.cs#L318-L327

Reply:
Reverted